platform/brains2/src/duo/main.cpp

```diff
@@ -286,24 +284,21 @@ static void main_loop(){
     DatoUSB::background_update();
 
     if (power_check()) {
-      keyboard_to_note();
-      pitch_update(); // ~30us
-      synth_update(); // ~ 100us
-      midi_send_cc();
-      Drums::update(); // ~ 700us
-      midi_handle();
-      sequencer_update();
-      headphone_jack_check();
-      #ifdef DEV_MODE
-      if (!dfu_flag) {
-      #endif
-      led_update();
-      FastLED.show();
-      pots_read();
-      keys_scan(); // 14 or 175us (depending on debounce)
-      #ifdef DEV_MODE
+      if (millis() - frame_time > frame_interval) {
```

Comment:
The removal of the throttling is what caused the issue with the Env slider LED. I'm not sure exactly why, but I see no problem with adding it again for now (since this is what we have on `main` right now also), so won't dig deeper into it.

Reply:
There is actually an interesting explanation to this. The env slider led reads the peak audio value between function calls to determine its brightness. If you read the peak audio volume at a very fast rate you are actually sampling the audio signal, so not only getting the peaks but also the samples between peaks. Limiting the rate at which peak() is called to sub audio rates prevents this. 

There is a tradeoff: faster sampling leads to smoother fades but the brightness will be influenced by low frequency notes.